rclcpp/CMakeLists.txt

```diff
@@ -2,6 +2,11 @@ cmake_minimum_required(VERSION 3.12)
 
 project(rclcpp)
 
+option(DEFINE_CONTENT_FILTER "Content filter feature support" ON)
```

Comment:
IMO, we should remove this, as it's really not doing anything

Reply:
This is a CMake option to control whether to build the CFT or not.
If building `rclcpp` without CFT and its test cases,  users could use `colcon build --cmake-args -DDEFINE_CONTENT_FILTER=OFF -DBUILD_TESTING=ON`, please notice that the test is also needed to control by the [option](https://github.com/ros2/rclcpp/blob/master/rclcpp/test/rclcpp/CMakeLists.txt#L737).

If removing this line, users need to comment on two locations([1](https://github.com/ros2/rclcpp/blob/master/rclcpp/CMakeLists.txt#L7), [2](https://github.com/ros2/rclcpp/blob/master/rclcpp/test/rclcpp/CMakeLists.txt#L738-L754)) if they want to build `rclcpp` without CFT.

Reply:
Anyway, let me create a new PR to remove it.

Reply:
[the follow up PR](https://github.com/ros2/rclcpp/pull/1914) was created.

Reply:
as @iuhilnehc-ynos mentioned, we are okay to get rid of this option. but i thought that this was something we discussed and wanted to have as an option to let the user application if they want to use the ContentFilteredTopic feature?

Reply:
Yes, we had discussed something like that, but this doesn't really accomplish that (we were wanting to try and achieve ABI protection not just turning the feature on/off). So, I think it's better we just remove this and document the limitations it has depending on the rmw you choose.